src/main/java/tools/jackson/databind/util/JacksonCollectors.java

```diff
@@ -0,0 +1,38 @@
+package tools.jackson.databind.util;
+
+import java.util.stream.Collector;
+import tools.jackson.databind.JsonNode;
+import tools.jackson.databind.node.ArrayNode;
+import tools.jackson.databind.node.JsonNodeFactory;
+
+/**
+ * Utility class that provides custom {@link Collector} implementations to support Stream operations.
+ * <p>
+ * This class is not meant to be instantiated and serves only as a utility class.
+ * </p>
+ */
+public class JacksonCollectors {
+
+    private JacksonCollectors() {
+        throw new UnsupportedOperationException("Utility class cannot be instantiated.");
+    }
+
+    /**
+     * Creates a {@link Collector} that collects {@link JsonNode} elements into an {@link ArrayNode}.
+     * <p>
+     * This method uses a {@link JsonNodeFactory} to create an empty {@link ArrayNode} and then adds each
+     * {@link JsonNode} to it.
+     * </p>
+     *
+     * @return a {@link Collector} that collects {@link JsonNode} elements into an {@link ArrayNode}
+     */
+    public static Collector<JsonNode, ArrayNode, ArrayNode> toJsonNode() {
+        final JsonNodeFactory jsonNodeFactory = new JsonNodeFactory();
```

Comment:
Although for most use the default `JsonNodeFactory` is fine, I think it'd better to have a variant that takes in `JsonNodeFactory` (or `JsonNodeCreator` maybe, which it implements), and then this overload that uses

    JsonNodeFactory.instance

Reply:
Can you check now please?